docusaurus/docs/tutorials/build-an-app-plugin.md

```diff
@@ -0,0 +1,186 @@
+
+1. Create a new file called `src/pages/NewPage.tsx` and add the following code:
+
+   ```tsx title="src/pages/NewPage.tsx"
+   import React from 'react';
+   import { PluginPage } from '@grafana/runtime';
+
+   export function NewPage() {
+     return <PluginPage>New Page</PluginPage>;
+   }
+   ```
+
+1. Modify the routes in `src/components/App/App.tsx` to recognize the new page:
+
+   ```tsx title="src/components/App/App.tsx"
+   {
+     /* .... */
+   }
+   <Route path="new-page" element={<NewPage />} />;
+   ```
+
+1. Save the file.
+1. Reload Grafana to see the new page in place.
+
+You don't need to register all your pages inside `includes` in your `plugin.json`. Register only pages that you want to add to the navigation menu.
+
+:::note
+
+You can only have one level of pages in the navigation menu. Sub-menu items are not supported.
+
+:::
+
+## Configuration page
+
+You can add a configuration page to your App plugin where your users can store any settings your plugin might need. Your plugin should already have an example configuration page with its source code located in `src/components/AppConfig/AppConfig.tsx`.
+
+### Saving user settings
+
+To store user settings, send a POST request to `/api/plugins/%PLUGIN_ID%/settings` with the `jsonData` and `secureJsonData` as data.
+
+```ts
+export const updatePluginSettings = async (pluginId: string, data: Partial<PluginMeta>) => {
+  const response = await getBackendSrv().fetch({
+    url: `/api/plugins/${pluginId}/settings`,
+    method: 'POST',
+    data, // data: { jsonData: { ... }, secureJsonData: { ... } }
+  });
+
+  return lastValueFrom(response);
+};
+```
+
+### Retrieve user settings
+
+The user settings are part of the plugin `meta`. You can retrieve them inside a React component by using the `usePluginContext` hook. For example:
+
+```tsx
+import React from 'react';
+import usePluginContext from '@grafana/data';
+
+function MyComponent() {
+  const context = usePluginContext();
+  // user settings
+  const jsonData = context.meta.jsonData;
+}
+```
+
+## Fetch data from frontend code using the data proxy
+
+If you want to fetch data from your app frontend code without [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) issues or using authenticated requests, then you want to use the [data proxy](../create-a-plugin/extend-a-plugin/fetch-data-from-frontend.md).
+
+## Bundle plugins inside your app
+
+You can bundle data sources and panel plugins inside your app plugins. See [Work with nested plugins](../create-a-plugin/extend-a-plugin/nested-plugins).
+
+## Include external plugins
+
+If you want to let users know that your app requires an existing plugin, you can add it as a dependency in `plugin.json`. Note that they'll still need to install it themselves.
+
+```json title="src/plugin.json"
+"dependencies": {
+  "plugins": [
+    {
+      "type": "panel",
+      "name": "Worldmap Panel",
+      "id": "grafana-worldmap-panel",
```

Comment:
Please pick a different, non deprecated plugin 😄

Reply:
My understanding was that plugins listed inside the dependencies were automatically installed by Grafana when you installed the "parent" plugin. Is this not the case?

Reply:
that is the case - good catch Tom